networkit/cpp/components/ConnectedComponents.cpp

```diff
@@ -21,6 +22,51 @@ ConnectedComponents::ConnectedComponents(const Graph& G) : G(G) {
     }
 }
 
+cc_result ConnectedComponents::get_raw_partition(const Graph & G) {
+    node n_components = 0;
+    node max_id = G.upperNodeIdBound();
+    auto mapping_array = new node[max_id];
```

Comment:
What's the difference between `mapping_array` and `component` in `ConnectedComponents::run()`?

Reply:
Or asked differently, why do we need to implement this method, rather than running `ConnectedComponents::run()` followed by `ConnectedComponents::getPartition()`? Seems like the only difference is that mapping sizes are also computed, but that could be a separate method.

Reply:
`ConnectedComponents::run()` creates a `std::vector` which holds the data we are interested in. The problem is that there is no way to steal the data from the container and move the ownership to python.
This is why I choose to basically copy the method and create a static method which creates a raw pointer, which then can be passed to python through the `ArrayWrapper` class.

Reply:
So we save one copy, right? Is that significant time / memory usage? If yes, please add a comment to explain why you copied the method :)

Reply:
Yes, the mapping_array is something like `(num_nodes * 8) GB` in size. I added a comment :)